gc.c

```diff
@@ -8209,6 +8010,171 @@ gc_marks_step(rb_objspace_t *objspace, size_t slots)
 }
 #endif
 
+static bool
+gc_compact_heap_cursors_met_p(rb_heap_t *heap)
+{
+    GC_ASSERT(heap->sweeping_page && heap->compact_cursor);
```

Comment:
This assertion is failing.
Assertion expressions should be kept simple, instead of combining by `&&`.